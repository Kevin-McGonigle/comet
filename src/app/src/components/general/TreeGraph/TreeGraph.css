.treeGraphContainer {
	display: flex;
	align-items: center;
}

div.custom-container {
	background-color: #242424;
}

.node circle {
	fill: #F3F3FF;
	stroke: #2593B8;
	stroke-width: 1.5px;
}

.node text {
	color: black;
}

.class {
	fill:#fd79a8;
	cursor: pointer;
}

.class text {
	font-size: 12px;
	fill: #F4F4F4;
	text-shadow: 0 1px 4px black;
}

.class-link {
	fill: none;
	stroke: #fd79a8;
	stroke-width: 1px;
}

.variable {
	fill: #55efc4;
	cursor: pointer;
}

.variable text {
	font-size: 12px;
	fill: #F4F4F4;
	text-shadow: 0 1px 4px black;
}

.variable-link {
	fill: none;
	stroke: #55efc4;
	stroke-dasharray: 3;
	stroke-width: 1px;
}

.node text {
	font-size: 12px;
	fill: #F4F4F4;
	text-shadow: 0 1px 4px black;
}

.node {
	cursor: pointer;
}

path.link {
	fill: none;
	stroke: #2593B8;
	stroke-width: 1px;
}