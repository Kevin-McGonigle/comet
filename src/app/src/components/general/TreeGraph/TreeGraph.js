import React from 'react';
import styles from './TreeGraph.css';
import classnames from 'classnames';
import Tree from 'react-tree-graph';

const cx = args => classnames(styles, args);

const TreeGraph = props => {
    const {
        tree,
    } = props;

    return (
        <div className={cx('treeGraphContainer')}>
            <Tree
                data={tree}
                height={900}
                width={800}
                svgProps={{transform: 'rotate(90)'}}
            />
        </div>
    );
};

export default TreeGraph;