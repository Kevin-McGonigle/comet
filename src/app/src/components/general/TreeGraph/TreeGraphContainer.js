import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import TreeGraph from './TreeGraph';
// import { selectors as treeSelectors } from '../../store/tree/selectors';
import { actions as treeActions } from '../../../store/tree/tree';

const mapStateToProps = state => ({
    tree: state.tree,
})

const mapDispatchToProps = dispatch => bindActionCreators(treeActions, dispatch)


export const TreeGraphContainer = connect(mapStateToProps, mapDispatchToProps)(TreeGraph)



