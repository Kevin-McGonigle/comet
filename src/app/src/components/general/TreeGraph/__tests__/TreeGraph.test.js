import React from 'react';
import TreeGraph from '../TreeGraph';
import initialState from '../../../../store/tree/tree';
import { shallow } from 'enzyme';
import "../../../../setupTests"

describe('TreeGraph', () => {
    test('Should render TreeGraph succesfully', () => { 
        const component = shallow(<TreeGraph tree={initialState} />);
        expect(component).toMatchSnapshot();
    });
    test('Should call hideAllChildNodes on node click', () => {
        const hideAllChildNodes = jest.fn();
        const component = shallow(<TreeGraph tree={initialState} hideAllChildNodes={hideAllChildNodes} />);
        const node = component.find('#TreeGraph.js');
        node.simulate('click');
        expect(hideAllChildNodes).toHaveBeenCalled();
    });
})
