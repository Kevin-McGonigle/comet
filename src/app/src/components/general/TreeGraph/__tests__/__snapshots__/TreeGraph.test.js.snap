// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TreeGraph successfully 1`] = `"<div class=\\"treeGraphContainer\\"><svg transform=\\"rotate(90)\\" height=\\"900\\" width=\\"800\\"><path class=\\"link\\" d=\\"M10,513.3333333333333C115,513.3333333333333 115,440 220,440\\"></path><path class=\\"link\\" d=\\"M10,513.3333333333333C115,513.3333333333333 115,586.6666666666666 220,586.6666666666666\\"></path><path class=\\"class-link\\" d=\\"M220,440C325,440 325,220 430,220\\"></path><path class=\\"class-link\\" d=\\"M220,440C325,440 325,660 430,660\\"></path><path class=\\"variable-link\\" d=\\"M430,220C535,220 535,146.66666666666666 640,146.66666666666666\\"></path><path class=\\"variable-link\\" d=\\"M430,220C535,220 535,293.3333333333333 640,293.3333333333333\\"></path><path class=\\"link\\" d=\\"M430,660C535,660 535,586.6666666666666 640,586.6666666666666\\"></path><path class=\\"link\\" d=\\"M430,660C535,660 535,733.3333333333333 640,733.3333333333333\\"></path><g class=\\"node\\" transform=\\"translate(10, 513.3333333333333)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">/src</text></g><g class=\\"class\\" id=\\"TreeGraph.js\\" transform=\\"translate(220, 440)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">TreeGraph.js</text></g><g class=\\"variable\\" transform=\\"translate(220, 586.6666666666666)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">TreeGraph.css</text></g><g class=\\"variable\\" id=\\"arguments\\" transform=\\"translate(430, 220)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">arguments</text></g><g class=\\"node\\" transform=\\"translate(430, 660)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">functions</text></g><g class=\\"variable\\" id=\\"Int arg1\\" transform=\\"translate(640, 146.66666666666666)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">Int arg1</text></g><g class=\\"variable\\" id=\\"Int arg2\\" transform=\\"translate(640, 293.3333333333333)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">Int arg2</text></g><g class=\\"node\\" transform=\\"translate(640, 586.6666666666666)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">square -&gt; Int arg1, Int arg2</text></g><g class=\\"node\\" transform=\\"translate(640, 733.3333333333333)\\"><circle r=\\"5\\"></circle><text dx=\\"5.5\\" dy=\\"3.5\\">sum -&gt; Int arg1, Int arg2</text></g></svg></div>"`;
